extensions/panache/rest-data-panache/deployment/src/main/java/io/quarkus/rest/data/panache/deployment/methods/UpdatePatchMethodImplementor.java

```diff
@@ -0,0 +1,83 @@
+package io.quarkus.rest.data.panache.deployment.methods;
+
+import static io.quarkus.gizmo.MethodDescriptor.ofMethod;
+
+import javax.ws.rs.core.Response;
+
+import io.quarkus.gizmo.BranchResult;
+import io.quarkus.gizmo.BytecodeCreator;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.FieldDescriptor;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.rest.data.panache.deployment.ResourceMetadata;
+import io.quarkus.rest.data.panache.deployment.properties.ResourceProperties;
+import io.quarkus.rest.data.panache.deployment.utils.ResponseImplementor;
+
+public final class UpdatePatchMethodImplementor extends StandardMethodImplementor {
+
+    private static final String METHOD_NAME = "updatePatch";
+
+    private static final String RESOURCE_UPDATE_PATCH_METHOD_NAME = "updatePatch";
+
+    private static final String RESOURCE_UPDATE_METHOD_NAME = "update";
+
+    private static final String RESOURCE_GET_METHOD_NAME = "get";
+
+    private static final String REL = "updatePatch";
+
+    @Override
+    protected void implementInternal(ClassCreator classCreator, ResourceMetadata resourceMetadata,
+            ResourceProperties resourceProperties, FieldDescriptor resourceField) {
+        MethodCreator methodCreator = classCreator.getMethodCreator(METHOD_NAME, Response.class.getName(),
+                resourceMetadata.getIdType(), resourceMetadata.getEntityType());
+
+        // Add method annotations
+        addPathAnnotation(methodCreator,
+                appendToPath(resourceProperties.getPath(RESOURCE_UPDATE_PATCH_METHOD_NAME), "{id}"));
+        addTransactionalAnnotation(methodCreator);
+        addPatchAnnotation(methodCreator);
+        addPathParamAnnotation(methodCreator.getParameterAnnotations(0), "id");
+        addConsumesAnnotation(methodCreator, APPLICATION_JSON);
+        addProducesAnnotation(methodCreator, APPLICATION_JSON);
+        addLinksAnnotation(methodCreator, resourceMetadata.getEntityType(), REL);
+
+        // Invoke resource methods
+        ResultHandle resource = methodCreator.readInstanceField(resourceField, methodCreator.getThis());
+        ResultHandle id = methodCreator.getMethodParam(0);
+        ResultHandle entityToSave = methodCreator.getMethodParam(1);
+
+        BranchResult entityExists = doesEntityExist(methodCreator, resourceMetadata.getResourceClass(), resource, id);
+        updateAndReturn(entityExists.trueBranch(), resourceMetadata.getResourceClass(), resource, id, entityToSave);
+        createAndReturn(entityExists.falseBranch(), resourceMetadata.getResourceClass(), resource, id, entityToSave);
```

Comment:
Sorry that I missed this last time, but I don't think that PATCH should create a resource. I think it's only supposed to udpate an existing entity (different from PUT which creates or updates it).